pkg/fdbstatus/status_checks.go

```diff
@@ -162,12 +163,21 @@ func CanSafelyRemoveFromStatus(logger logr.Logger, client fdbadminclient.AdminCl
 		}
 	}
 
+	// Verify that all processes that are assumed to be fully excluded based on the machine-readable status are actually
+	// not serving any roles by running the exclude command again. If those processes are actually fully excluded and are not
+	// serving any roles, the exclude command should terminate quickly, otherwise we will hit a timeout, and we know that
+	// not all processes are fully excluded. This is meant to be an additional safeguard if the machine-readable status
+	// returns the wrong signals.
+	if len(exclusions.fullyExcluded) > 0 {
+		// When we hit a timeout error here we know that at least one of the fullyExcluded is still not fully excluded.
+		return notSafeToDelete, client.ExcludeProcesses(exclusions.fullyExcluded)
```

Comment:
CC @jzhou77 any concerns if we add the additional `exclude` command? The idea is to confirm that the processes that are assumed to be fully excluded based on the machine-readable, will be used in an additional `exclude` command. If the exclude command returns before it hits the timeout, those processes are actually excluded. If the command times out, we know that the machine-readable status was not correct and at least one process is still part of the key or log servers.

Reply:
Do we have any safeguards in place to prevent the death spirals we saw in the past with trying to check for exclusions while a cluster was in a bad state? I expect it's better to have this than not do the check and potentially remove processes before they are fully excluded, but I think we should think about finding a middle ground where we're not doing unnecessary exclude commands when the cluster is already in a bad shape.